src/Mesh/Interpolation.jl

```diff
@@ -0,0 +1,371 @@
+module Interpolation
+
+using MPI
+import GaussQuadrature
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.Mesh.Geometry
+using CLIMA.Mesh.Elements
+using CLIMA.Mesh.Tens
+using LinearAlgebra
+using StaticArrays
+
+export Interpolation_Brick, interpolate_brick!, interpolate_brick_mf!, Interpolation_Cubed_Sphere, invert_trilear_mapping_hex, interpolate_cubed_sphere!
+
+#--------------------------------------------------------
+
+# This interpolation routine works for a brick, where stretching/compression happens only along the x, y & z axis.
+# Here x = X(ξ1); y = Y(ξ2); z = Z(ξ3)
+
+struct Interpolation_Brick{FT <:AbstractFloat}
+    El::UnitRange{Int64}
```

Comment:
Might be a good idea to use the same names as in Topologies
```suggestion
    realelems::UnitRange{Int64}
```